web3/middleware/__init__.py

```diff
@@ -88,3 +89,26 @@ def combine_middlewares(
         reversed(middlewares),
         provider_request_fn,
     )
+
+
+async def async_combine_middlewares(
+    middlewares: Sequence[Middleware],
+    web3: 'Web3',
+    provider_request_fn: Callable[[RPCEndpoint, Any], Any]
+) -> Callable[..., RPCResponse]:
+    """
+    Returns a callable function which will call the provider.provider_request
+    function wrapped with all of the middlewares.
+    """
+    accumulator_fn = provider_request_fn
+    for middleware in reversed(middlewares):
+        accumulator_fn = await apply_middleware(middleware, accumulator_fn, web3)
+    return accumulator_fn
+
+
+async def apply_middleware(
+    middleware: Middleware,
```

Comment:
This name through me.  I assumed this was the actual middleware function, but it's the "middleware constructor"?  No direct suggestion on naming but it might be worth establishing conventions for these two so that they can be logically separated.

Reply:
whoops, you're right. Good suggestion!